### PR TITLE
SEO and performance improvements

### PR DESCRIPTION
**Pull request by mairaw**
This PR addresses most of the SEO issues identified for this site. @v-mepa can you please take a look at the meta descriptions I added?

@BethMassi I also saw some issues with the terms page and I'm suggesting some improvements. Can you take a look to see if it's ok? It's the last file on this PR

Contributes to https://github.com/dotnet/website/issues/3510

Issues not addressed here:
- low content
- 301 redirect (couldn't repro)

### CODE REVIEW

File: _includes/header.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <header class="site-header">
   <div class="container">
-    <h1><a href="{{ '/' | relative_url }}">Mod dotnet-bot</a></h1>
+    <a style="color:white;font-weight:600;font-size:20px;margin-right:48px" href="{{ '/' | relative_url }}">Mod dotnet-bot</a>
```
Comment (mairaw) [inline _includes/header.html:3]: @danzhu54 can you check if you can move this to the proper file? I tried but it didn't seem to work so added here manually.

**Comment by BethMassi** [comment]
All looks good to me!

**Comment by v-mepa** [comment]
Thanks, @mairaw for downsizing the images and updating page titles and meta. Looks good to me.

**Review by Arhell — APPROVED** [review]

File: _includes/header.html (hunk 1 above)
Comment (danzhu54) [reply _includes/header.html:3]: We should probably move this to https://github.com/dotnet-foundation/mod-dotnet-bot/blob/32808d87b4cb76200622753715d657bd9e7c92af/_dev/scss/layout/_header.scss. I can add it.

Comment (mairaw) [reply _includes/header.html:3]: I think that's where I tried and it didn't work. But maybe I did something wrong 🤷🏻‍♀️

Comment (danzhu54) [reply _includes/header.html:3]: The gulp task is supposed to compile the sass into main.css under ./assets/css but I'm running into a compilation issue at the moment. We we can merge these changes for now.

**Pull request merged**